.github/workflows/build.yml

```diff
@@ -1,7 +1,7 @@
 on: [ push, pull_request ]
 
 env:
-  CFLAGS: -Werror -Wall -Wextra -Wno-error=sign-compare -Wno-unused-parameter -Wno-error=missing-field-initializers
+  CFLAGS: -Wall -Wextra -Wno-unused-parameter -Wformat=2 -Wformat-overflow=2 -Wformat-signedness -Wformat-truncation=2 -Wshift-overflow=2 -Wstrict-overflow=2 -Wswitch-enum -Werror -Wno-error=missing-field-initializers
```

Comment:
Follow-up here because I just noticed this in libwacom: in meson `CFLAGS` are **appended** to the cflags specified in `meson.build`. So in the simple case here with `-Wall -Wextra` the actually used sequence ends up being:
```
clang -Wall -Wextra -Wno-foo -Wall -Wextra -Wno-error=sign-compare ...`
```
With the first set being the `meson.build` ones including the specific ones set there and then the set from `CFLAGS`. Unfortunately the second `-Wall -Wextra` re-enables all, ignoring the previous `-Wno-foo` ones. Which is why we need `-Wno-unused-parameter` in here despite that being listed in `meson.build`.

See https://github.com/linuxwacom/libwacom/pull/478

Reply:
I didn't realize that we were setting CFLAGS for meson builds. That explains some things, and I agree is probably the better way to go. I think the only thing it makes sense to leave in the Github action is `-Werror` as a sanity check; all the others should be noisy when actually doing development.

Reply:
I had a go at splitting this up in the github workflow but ... urgh, that workflow API is terrible. You can set a global `env` but you can't reference that in `job`. You *can* reference it in `jobs.step`, but then you must not forget it in any step that may use it. Like this:

```yaml
env:
   AM_CFLAGS: -Werror -Wfoo

jobs:
  autotools-job:
    env:
      FOO: bar  # this is allowed
      CFLAGS: ${{env.AM_CFLAGS}} # but this is not
    steps:
    - run: ./configure
      env:
        CFLAGS: ${{env.AM_CFLAGS}} # this works
    - run: make
      # oops, now you forgot to specify the env
```
So after wasting over an hours on this... let's go with your solution ;)